packages/calcite-components/src/components/text-area/text-area.tsx

```diff
@@ -260,12 +260,12 @@ export class TextArea
     return (
       <Host>
         <textarea
-          aria-invalid={toAriaBoolean(this.value?.length > this.maxLength)}
+          aria-invalid={toAriaBoolean(this.isCharacterLimitExceeded())}
```

Comment:
To mitigate with NVDA and FF, add in an `aria-describedby` to provide additional context.